### ISSUE TITLE
Why your way to get quantized vectors is so tortuous?

### ISSUE THREAD

**Issue by voidism** [MishaLaskin/vq-vae#2]
Thanks for your awesome VQ-VAE implementation!

I read the code [here](https://github.com/MishaLaskin/vqvae/blob/d761a999e2267766400dc646d82d3ac3657771d4/models/quantizer.py#L55-L60), which is for getting quantized vectors:

https://github.com/MishaLaskin/vqvae/blob/d761a999e2267766400dc646d82d3ac3657771d4/models/quantizer.py#L55-L60

I am wondering why don't you just use the forward function of the `nn.Embedding` object `self.embedding`, like this:

```
min_encoding_indices = torch.argmin(d, dim=1)
z_q = self.embedding(min_encoding_indices)
```
(The two ways can get the same results.)

If I change the code like this, will I get into some trouble? (unable to pass gradient... etc)

Thank you so much!

**Comment by loongzeng** [comment]
I have the same question...
